### PR TITLE
Allow over-ridable renderer for Item and Composite views at an instance level.

### PR DESCRIPTION
**Pull request by jshthornton**

### CODE REVIEW

**Comment by jasonLaster** [comment]
We've discussed this in the past, but I think this makes sense.

**Comment by jasonLaster** [comment]
At a minimum we can add a getRenderer that wraps Mn.renderer

**Comment by ahumphreys87** [comment]
yeah, I'd prefer `getRenderer` for consistency.

@jshthornton could you remove the changes in the built versions please?

**Comment by paulfalgout** [comment]
`getRenderer` seems like it could just be added to the base `View` as well

**Comment by jshthornton** [comment]
I agree with this. I might be able to find the time to contribute the changes.

**Comment by jshthornton** [comment]
@ahumphreys87 I need to use this fork as production until this PR is applied, therefore I need the build files to accompany them.

**Comment by jshthornton** [comment]
Changes have been applied. Awaiting Travis.

File: src/view.js [hunk 1]
```diff
@@ -349,6 +349,11 @@ Marionette.View = Backbone.View.extend({
     });
   },
 
+  // Returns the view's renderer.
+  getRenderer: function() {
```
Comment (samccone) [inline src/view.js]: we need to add docs about this

File: src/item-view.js [hunk 2]
```diff
@@ -85,7 +84,7 @@ Marionette.ItemView = Marionette.View.extend({
     var data = this.mixinTemplateHelpers(this.serializeData());
 
     // Render and add to el
-    var html = Marionette.Renderer.render(template, data, this);
+    var html = (this.getRenderer()).render(template, data, this);
```
Comment (samccone) [inline src/item-view.js]: We should user our getOption helper here to get this value

File: src/composite-view.js [hunk 3]
```diff
@@ -92,7 +91,7 @@ Marionette.CompositeView = Marionette.CollectionView.extend({
     this.triggerMethod('before:render:template');
 
     var template = this.getTemplate();
-    var html = Marionette.Renderer.render(template, data, this);
+    var html = (this.getRenderer()).render(template, data, this);
```
Comment (samccone) [inline src/composite-view.js]: We should user our getOption helper here to get this value

File: src/item-view.js [hunk 4]
```diff
@@ -5,7 +5,6 @@
 // with underscore.js templates, serializing the view's model or collection,
 // and calling several methods on extended views, such as `onRender`.
 Marionette.ItemView = Marionette.View.extend({
-
```
Comment (samccone) [inline src/item-view.js]: :x: whitespace change

File: src/composite-view.js [hunk 5]
```diff
@@ -7,7 +7,6 @@
 // Extends directly from CollectionView and also renders an
 // a child view as `modelView`, for the top leaf
 Marionette.CompositeView = Marionette.CollectionView.extend({
-
```
Comment (samccone) [inline src/composite-view.js]: :x: whitespace change

**Comment by jasonLaster** [comment]
@jshthornton mind re-opening PR against `next` also you don't need to do a build or anything, just your patch :)

**Comment by ianmstew** [comment]
@jshthornton Can you remove the build assets from the commit?  `lib/*`

**Comment by ahumphreys87** [comment]
Closed in favour of #2682

**Pull request closed without merging**